### DISCUSSION TITLE
Duplicate the Existing Collections and Records with flutter

### DISCUSSION THREAD

**Discussion by alouis19** [pocketbase/pocketbase#3939] [Q&A] [answered]
Hello everyone, 
I have a small question, I would like to duplicate a recording from a collection with flutter. I can't find a copy() or duplicate() function in the docs. Do any of you have an idea to help me?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no such builtin method at the moment in the SDKs or the web APIs because the duplication will differ depending on what actually you want to duplicate (eg. duplicate only the main record fields, files, cascade the duplicate operation to its related records, etc.; see for example https://github.com/pocketbase/pocketbase/discussions/3812).

In general, if you want only to duplicate the main record fields client-side you'll have to:
- fetch an existing object
- clear its id value
- clear its `file` fields values OR if you want to duplicate the files, you'll have to also download them and reupload them
- submit a new create request with the modified data


Another alternative for a duplicate action could be to register a custom route with [Go](https://pocketbase.io/docs/go-routing/) or [JS](https://pocketbase.io/docs/js-routing/) and perform the required db and file operations server-side but there are not enough details in your question to help you with this.

If you need further help, please elaborate on your use case, collections structure (_it can be exported from Admin UI > Settings > Export collections_) and the desired duplication behavior.

**Reply by alouis19** [reply]
Thank you very much for your prompt reply. I had thought of this solution but I wanted to know if a function already existed to duplicate with images.
